airflow/utils/types.py

```diff
@@ -42,7 +43,11 @@ class DagRunType(str, enum.Enum):
     def __str__(self) -> str:
         return self.value
 
-    def generate_run_id(self, logical_date: datetime) -> str:
+    def generate_run_id(self, logical_date: datetime | None, run_after: datetime | None) -> str:
```

Comment:
```suggestion
    def generate_run_id(self, *, logical_date: datetime | None, run_after: datetime | None) -> str:
```

same here

Reply:
I have changed it here: https://github.com/apache/airflow/pull/46616